src/focus-visible.js

```diff
@@ -123,64 +126,44 @@ function init() {
    * @param {Event} e
    */
   function onBlur(e) {
-    if (e.target == document) {
+    if (e.target == document || e.target.nodeName == 'HTML') {
       return;
     }
 
-    removeFocusRingClass(e.target);
-  }
-
-  /**
-   * When the window regains focus, restore the focus-visible class to the element
-   * to which it was previously applied.
-   */
-  function onWindowFocus() {
-    // When removing the activeElement from DOM it's possible IE11 is in state
-    // document.activeElement === null
-    if (!document.activeElement) {
-      return;
+    if (e.target.classList.contains('focus-visible')) {
+      hadFocusVisibleRecently = true;
```

Comment:
Suggested comment here:

To detect a tab switch, we look for a blur followed rapidly by a visibility change. If we don't see a visibility change within 100ms, it's probably a regular focus change.